design/crd-status.md

```diff
@@ -0,0 +1,237 @@
+of the resource _per peer / node_, which exposes the state of the session between
+the speaker running on a given node and a given peer.
+
+The name of a given instance will be like `nodename-peer`, and each instance will
+be labeled with the name of the node and the peer it refers to, to make it easier
+to list the status of all the sessions related to a given `BGPPeer` and a given
+node.
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: BGPSessionStatus
+metadata:
+  name: worker0-peer1
+  namespace: metallb-system
+  labels:
+    metallb.io/node: worker0
+    metallb.io/peer: peer1
+status:
+    bgpStatus: Established
+    bfdStatus: Up
+```
+The string exposed is taken directly from the output of FRR.
+
+If BFD is not configured for a given `BGPPeer`, the exposed bfdStatus will be "N/A".
+
+#### Service Announcement - BGP
+
+Given a service and a node, we want to expose the `BGPpeer`s it is advertised to.
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisementStatus
+metadata:
+  name: service1-worker0
+  namespace: metallb-system
+  labels:
+    metallb.io/node: worker0
+    metallb.io/service: service1
+status:
+    bgpPeers:
+        - peerA
+        - peerB
+```
+
+The labels will allow to easily discover which services are advertised from
+a given node, or all the peers a given service is advertised to.
+
+#### Service Announcement - L2
+
+The useful information related to L2 are related to the node that is exposing
+the service, and via which interfaces.
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2AdvertisementStatus
+metadata:
+  name: service1-worker0
+  namespace: metallb-system
+  labels:
+    metallb.io/node: worker0
+    metallb.io/service: service1
+status:
+    node: worker0
+    interfaces:
+        - eth0
+        - eth1
+```
+
+The absence of interfaces means all interfaces are selected.
+
+## Alternatives
+
+We might consider exposing all the information (or part of it) as labels to be applied
+to the services. However, the current proposal is easier to query and to navigate,
+thanks to the various labels applied.
+
+On top of that, there is some information (such as the state of the sessions) that
+can't be added as service annotations.
+
+## Development Phases
+
+Each CRD can be developed and exposed regardless of the others. In order to consider
+the status of a given CRD complete, the following items must be finished:
+
+* implementation
+* e2e testing
+* documentation
+
+After this proposal converges and is accepted, separate issues will be filed in order
+to ease the development and allow the development to move in parallel.
+
+### Test Plan
+
+**e2e tests**: E2E tests will be expanded to cover bfd, using external container(s)
+running FRR. The test will need to cover the cases where a node is dropped, verifying
+that the broken route is detected by BFD.
+We **must** ensure through tests that unnecessary updates are not generated if the
+exposed status does not change.
```

Comment:
I guess you mean having a BGP session backed up with BFD.
Plus, we should also mention covering the L2 Advertisement.

Reply:
ah damn. This is a leftover, I copied it from the bfd proposal :sweat_smile: